packages/manager/apps/vrack-services/src/components/EditableText.tsx

```diff
@@ -0,0 +1,120 @@
+import React from 'react';
+import { OsdsIcon } from '@ovhcloud/ods-components/icon/react';
+import { ODS_ICON_NAME, ODS_ICON_SIZE } from '@ovhcloud/ods-components/icon';
+import { OsdsButton } from '@ovhcloud/ods-components/button/react';
+import { OsdsInput } from '@ovhcloud/ods-components/input/react';
+import {
+  ODS_INPUT_SIZE,
+  ODS_INPUT_TYPE,
+  OdsInputValueChangeEvent,
+} from '@ovhcloud/ods-components/input';
+import {
+  ODS_BUTTON_SIZE,
+  ODS_BUTTON_TYPE,
+  ODS_BUTTON_VARIANT,
+} from '@ovhcloud/ods-components/button';
+import { ODS_THEME_COLOR_INTENT } from '@ovhcloud/ods-common-theming';
+
+export type EditableTextProps = React.PropsWithChildren<{
+  type?: ODS_INPUT_TYPE;
+  disabled?: boolean;
+  defaultValue: string;
+  onEditSubmitted: (newValue: string) => Promise<void>;
+}>;
+
+export type EditStatus = 'display' | 'editing' | 'loading';
+
+export const EditableText: React.FC<EditableTextProps> = ({
+  children,
+  defaultValue,
+  disabled,
+  type = ODS_INPUT_TYPE.text,
+  onEditSubmitted,
+}) => {
+  const [editStatus, setEditStatus] = React.useState('display' as EditStatus);
+  const [value, setValue] = React.useState(defaultValue);
+  const submitButton = React.useRef<HTMLButtonElement>();
+  const input = React.useRef<any>();
+
+  React.useEffect(() => {
+    if (editStatus === 'editing') {
+      setTimeout(() => {
+        input.current?.setFocus();
+      });
+    }
+  }, [editStatus]);
+
+  if (['editing', 'loading'].includes(editStatus)) {
+    return (
+      <form
+        onSubmit={async (event) => {
+          event.preventDefault();
```

Comment:
Why you need to prevent ? There is side action triggered ?

Reply:
Yes it reloads the page with the parameter by default
that's the default behavior of a form